scripts/prepare_sector_network.py

```diff
@@ -2264,56 +2264,66 @@ def add_heat(
                 )
 
         if options["tes"]:
-            n.add("Carrier", f"{heat_system} water tanks")
-
-            n.add(
-                "Bus",
-                nodes + f" {heat_system} water tanks",
-                location=nodes,
-                carrier=f"{heat_system} water tanks",
-                unit="MWh_th",
-            )
 
-            n.add(
-                "Link",
-                nodes + f" {heat_system} water tanks charger",
-                bus0=nodes + f" {heat_system} heat",
-                bus1=nodes + f" {heat_system} water tanks",
-                efficiency=costs.at["water tank charger", "efficiency"],
-                carrier=f"{heat_system} water tanks charger",
-                p_nom_extendable=True,
-            )
-
-            n.add(
-                "Link",
-                nodes + f" {heat_system} water tanks discharger",
-                bus0=nodes + f" {heat_system} water tanks",
-                bus1=nodes + f" {heat_system} heat",
-                carrier=f"{heat_system} water tanks discharger",
-                efficiency=costs.at["water tank discharger", "efficiency"],
-                p_nom_extendable=True,
-            )
+            n.add("Carrier", f"{heat_system} water tanks")
 
             tes_time_constant_days = options["tes_tau"][
                 heat_system.central_or_decentral
             ]
 
             n.add(
-                "Store",
+                "StorageUnit",
                 nodes + f" {heat_system} water tanks",
-                bus=nodes + f" {heat_system} water tanks",
-                e_cyclic=True,
-                e_nom_extendable=True,
+                bus=nodes + f" {heat_system} heat",
                 carrier=f"{heat_system} water tanks",
+                efficiency_store=costs.at[
+                    heat_system.central_or_decentral + " water tank charger",
+                    "efficiency",
+                ],
+                max_hours=costs.at[
+                    heat_system.central_or_decentral + " water tank storage",
+                    "energy to power ratio",
+                ],
+                efficiency_dispatch=costs.at[
+                    heat_system.central_or_decentral + " water tank discharger",
+                    "efficiency",
+                ],
+                p_nom_extendable=True,
                 standing_loss=1 - np.exp(-1 / 24 / tes_time_constant_days),
                 capital_cost=costs.at[
```

Comment:
Capital costs are related to power not to energy capacity --> Conversion from €/MWh to €/MW using energy to power ratio.